simphony/io/file_lattice.py

```diff
@@ -0,0 +1,194 @@
+
+    Returns
+    ----------
+    FileLattice
+        The lattice newly added to the file or existing in the file.
+
+    """
+    def __init__(self, file, name, type, base_vect, size, origin,
+                 record=None):
+        # Construct a default FileLattice
+        super(FileLattice, self).__init__(
+            name, type, base_vect, size, origin)
+        self._file = file._file
+        self._group = file._file.root.lattice
+        self._record = record
+
+        # If Lattice not in file, create a lattice
+        if self.name not in self._file.root.lattice:
+            if self._record is None:
+                # If record not specified use NoUIDRecord
+                self._record = NoUIDRecord
+            self._table = IndexedDataContainerTable(
+                self._group, name, self._record, np.prod(self._size))
+            for i in xrange(np.prod(self._size)):
+                self._table.append(DataContainer())
+
+            # Set table attributes
+            lattice = self._group._f_getChild(self.name)
+            lattice.attrs.type = self._type
+            lattice.attrs.base_vect = self._base_vect
+            lattice.attrs.size = self._size
+            lattice.attrs.origin = self._origin
+
+        # If Lattice already in file, return a reference
+        else:
+            self._table = IndexedDataContainerTable(self._group, self.name)
+            lattice = self._group._f_getChild(self.name)
+            self._type = lattice.attrs.type
+            self._base_vect = lattice.attrs.base_vect
+            self._size = lattice.attrs.size
+            self._origin = lattice.attrs.origin
+
+    def get_node(self, index):
+        """Get a copy of the node corresponding to the given index.
+
+        Parameters:
+        -----------
+        index: tuple of D x int (node index coordinate)
+
+        Returns:
+        -----------
+        A reference to a LatticeNode object
+
+        """
+        n = self._get_row_number(index)
+        dcs = {}
+        colnum = 0
+        for column in xrange(self._table._table.description.mask.shape[0]):
+            if self._table._table[n][0][column]:
+                data = self._table._table.description.data
+                key = eval(CUBA.__name__+'.'+data._v_names[colnum].upper())
+                dcs[key] = self._table[n][key]
+            colnum += 1
+        return LatticeNode(index, dcs)
+
+    def update_node(self, node, keywords=None):
+        """ Updates FileLattice data for a LatticeNode
+
+        Parameters:
+            node : LatticeNode
+                reference to LatticeNode object
+            keywords : list, optional
+                List of CUBA-keywords that are updated. If empty list
+                then all DC's are from the node
+
+        """
+        # Find correct row for node
+        n = self._get_row_number(node.index)
+        colnms = self._table._table.description.data._v_names
+        if keywords is None:
```

Comment:
node.data might have additional CUBA-keys defined with respect to FileLattice._table, so those have to filtered away before copying.